common/web/types/src/keyboard-object.ts

```diff
@@ -0,0 +1,198 @@
+/*
+ * Keyman is copyright (C) SIL Global. MIT License.
+ */
+import { OutputTarget } from './outputTarget.interface.js';
+import { TouchLayoutPlatform as LayoutFormFactorSpec } from './keyman-touch-layout/keyman-touch-layout-file.js';
+
+export type ComplexKeyboardStore = (string | { t: 'd', d: number } | { ['t']: 'b' })[];
+
+type KeyEvent = {};
+
+/**
+ * Stores preprocessed properties of a keyboard for quick retrieval later.
+ */
+export class CacheTag {
+  stores: { [storeName: string]: ComplexKeyboardStore };
+
+  constructor() {
+    this.stores = {};
+  }
+}
+
+export interface EncodedVisualKeyboard {
+  /** Represents CSS font styling to use for VisualKeyboard text */
+  F: string;
+  /** Should there be a 102nd key? */
+  K102?: boolean,
+  /**
+   * Keyboard Layer Specification: an object-based map of layer name to the keycaps for its
+   * 65 keys.  The 65 keys are ordered from left to right, then top to bottom.
+   *
+   * The key ID corresponding to each index of the array is specified within `Codes.dfltCodes`.
+   * Entries corresponding to `K_*` in `Codes.dfltCodes` are reserved for future use.
+   */
+  KLS?: { [layerName: string]: string[] },
+  /**
+   * @deprecated
+   * The older form for data in KLS - defines keycaps for 'default' keys, then 'shift' keys,
+   * in a single concatenated array.
+   */
+  BK?: string[];
+}
+
+export type LayoutSpec = {
+  "desktop"?: LayoutFormFactorSpec,
+  "phone"?: LayoutFormFactorSpec,
+  "tablet"?: LayoutFormFactorSpec
+}
+
+export type KeyboardObject = {
+  /**
+   * Used internally by Keyman Engine for Web to hold preprocessed stores.
+   */
+  _kmw?: CacheTag;
```

Comment:
I think this should be excluded from the `KeyboardObject` definition and be included as part of a subtype within /web source. (`type KmwKeyboardObject = KeyboardObject & { _kmw?: CacheTag };` or something like that.)